src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -424,18 +431,18 @@ private void AddLexicalClassifications(IClassificationService classificationServ
                 using var _ = Classifier.GetPooledList(out var tempList);
 
                 // If we have a syntax root ready, use the direct, non-async/non-blocking approach to getting classifications.
-                if (root == null)
-                    classificationService.AddSyntacticClassificationsAsync(document, span.Span.ToTextSpan(), tempList, cancellationToken).Wait(cancellationToken);
+                if (!lastProcessedDocumentOrRoot.TryGetFirst(out var root))
+                    classificationService.AddSyntacticClassificationsAsync(lastProcessedDocumentOrRoot.Second, span.Span.ToTextSpan(), tempList, cancellationToken).Wait(cancellationToken);
                 else
-                    classificationService.AddSyntacticClassifications(document.Project.Solution.Services, root, span.Span.ToTextSpan(), tempList, cancellationToken);
+                    classificationService.AddSyntacticClassifications(solutionServices, root, span.Span.ToTextSpan(), tempList, cancellationToken);
```

Comment:
nit, either consider flipping the if/else, or just do TryGetSecond in the above.  but i don't feel strongly, something just minorly confused me here.